src/client/theme-default/components/VPLocalSearchBox.vue

```diff
@@ -182,14 +181,22 @@ debouncedWatch(
         })
         const div = document.createElement('div')
         app.mount(div)
-        const sections = div.innerHTML.split(headingRegex)
+        const $ = cheerio.load(div.innerHTML)
         app.unmount()
-        sections.shift()
-        for (let i = 0; i < sections.length; i += 3) {
-          const anchor = sections[i + 1]
-          const html = sections[i + 2]
-          map.set(anchor, html)
-        }
+        const headings = $('h1, h2, h3, h4, h5, h6')
+        headings.each((_, el) => {
+          const anchor = $(el).find('a').attr('href')?.slice(1)
+          let next = $(el).next()
+          let html = ''
+          while (next.length && next[0].name !== el.name) {
+            const nonHTagHtml = `<${next[0].name} class="${
+                next[0].attribs.class
+            }">${next.html()}</${next[0].name}>`
```

Comment:
what's the reason behind selecting only class here?

Reply:
Since class is separated by cheerio, the elements with class will lose their class style if they are not written this way.